examples/md-flexible/Thermostat.h

```diff
@@ -12,120 +12,78 @@
+  // kinetic energy times 2
+  double kineticEnergyMul2 = 0;
+#ifdef AUTOPAS_OPENMP
+#pragma omp parallel reduction(+ : kineticEnergyMul2)
+#endif
+  for (auto iter = autopas.begin(); iter.isValid(); ++iter) {
+    auto vel = iter->getV();
+    kineticEnergyMul2 +=
+        particlePropertiesLibrary.getMass(iter->getTypeId()) * autopas::utils::ArrayMath::dot(vel, vel);
   }
+  // AutoPas works always on 3 dimensions
+  constexpr unsigned int dimensions{3};
+  return kineticEnergyMul2 / (autopas.getNumberOfParticles() * dimensions);
+}
 
-  /**
-   * Initial temperature + deltaTemp * simstep / n_thermostat.
-   */
-  ThermostatFloatType _tInit;
-
-  /**
-   * Target temperature.
-   */
-  const ThermostatFloatType _tTarget;
-
-  /**
-   * Temperature difference per thermostat application until t_target is reached.
-   */
-  const ThermostatFloatType _deltaTemp;
-
-  /**
-   * ParticlePropertiesLibrary to access Mass of Particles
-   */
-  ParticlePropertiesLibraryTemplate _particlePropertiesLibrary;
-};
-template <class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
-void Thermostat<AutoPasTemplate, ParticlePropertiesLibraryTemplate>::addBrownianMotion(AutoPasTemplate &autopas,
-                                                                                       bool useCurrentTemp) {
+/**
+ * Adds brownian motion to the given system.
+ *
+ * If useCurrentTemp is set to true the factor of the brownian motion is calculated per particle based on its mass and
+ * the system's temperature. Otherwise a constant factor of 0.1 is used.
+ * Set this to false if the system is initialized without velocities.
+ *
+ * @tparam AutoPasTemplate Type of AutoPas Object (no pointer)
+ * @tparam ParticlePropertiesLibraryTemplate Type of ParticlePropertiesLibrary Object (no pointer)
+ * @param autopas
+ * @param particlePropertiesLibrary
+ * @param useCurrentTemp
+ */
+template<class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
+void addBrownianMotion(AutoPasTemplate &autopas,
+                       ParticlePropertiesLibraryTemplate &particlePropertiesLibrary,
+                       bool useCurrentTemp) {
   // factors for the brownian motion per particle type.
-  std::map<size_t, ThermostatFloatType> factors;
+  std::map<size_t, double> factors;
   if (useCurrentTemp) {
     // brownian motion with disturbance depending on current temperature and mass
-    constexpr ThermostatFloatType boltzmannConst = 1.38064852e-23;  // Boltzmann constant in J/K
-    ThermostatFloatType currentTempMulKB = calcTemperature(autopas) * boltzmannConst;
-    for (auto typeID : _particlePropertiesLibrary.getTypes()) {
-      factors.emplace(typeID, std::sqrt(currentTempMulKB / _particlePropertiesLibrary.getMass(typeID)));
+    constexpr double boltzmannConst = 1.38064852e-23;  // Boltzmann constant in J/K
+    double currentTempMulKB = calcTemperature(autopas, particlePropertiesLibrary) * boltzmannConst;
```

Comment:
for a correct thermostat, you need a separate temperature for every component, otherwise, the lightest component will absorb all the energy.

Reply:
here you anyways shouldn't calculate a temperature, so not so important here